Formula/linkerd2.rb

```diff
@@ -0,0 +1,43 @@
+class Linkerd2 < Formula
+  desc "Linkerd configuration command-line utility"
+  homepage "https://linkerd.io"
+
+  url "https://github.com/linkerd/linkerd2.git",
+    :tag      => "stable-2.2.0",
+    :revision => "1cde1f083c5f245170f75e1974f3b20f7fa34d9c"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["CI_FORCE_CLEAN"] = "1"
+
+    srcpath = buildpath/"src/github.com/linkerd/linkerd2"
+    srcpath.install buildpath.children - [buildpath/".brew_home"]
+
+    cd srcpath do
+      system "bin/build-cli-bin"
+      bin.install "target/cli/darwin/linkerd"
+
+      # Install bash completion
+      output = Utils.popen_read("#{bin}/linkerd completion bash")
+      (bash_completion/"linkerd").write output
+
+      # Install zsh completion
+      output = Utils.popen_read("#{bin}/linkerd completion zsh")
+      (zsh_completion/"linkerd").write output
+
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    run_output = shell_output("#{bin}/linkerd 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
@fxcoudert it would need a k8s cluster to test functionality for it. Not sure how can I achieve it up, if there are any pointers it would be really helpful.

I tried to take reference from : https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb but was of not of much help.

Reply:
@fxcoudert for testing it, i would require k8s cluster to check functionality. I took references from  https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb, and this is also similar to it.

Reply:
Can we try to test some functionality, and check that it fails, then?

Reply:
Hi @fxcoudert, I've been working with @murarisumit on getting this into Homebrew. I agree on the need to validate `linkerd` installed correctly. I believe the `shell_output("#{bin}/linkerd version --client 2>&1")` command a few lines below satisfies that use case. The output of `linkerd version` is dependent on how the binary was compiled (and against what revision). Passing that test validates the `linkerd` binary was compiled and installed correctly. As @murarisumit mentioned, any further tests are dependent on access to a Kubernetes cluster (very similar to `brew install kubectl`) where we can't predict whether a command would succeed or fail.

Reply:
On our CI machines, it will always fail, so you can assume that.

Reply:
@fxcoudert We added a subsequent check that runs `linkerd install`. This command both exercises additional functionality, and succeeds regardless of whether Kubernetes is present. Let us know whether that is sufficient.